### DISCUSSION TITLE
Clarifying data retention

### DISCUSSION THREAD

**Discussion by dariom** [svix/svix-webhooks#1285] [Q&A] [answered]
I'd like to clarify my understanding of how data retention works in Svix.

On the Professional plan, there's a 90-day data retention. In your API reference, for example, for the [List Messages operation](https://api.svix.com/docs#tag/Message/operation/v1.message.list), you state:

> Note that by default this endpoint is limited to retrieving 90 days' worth of data relative to now... If you require data beyond those time ranges, you will need to explicitly set the before or after parameter as appropriate.

On the Professional plan, is it possible to get data older than 90 days old (using before/after parameters) or is that text intended for Enterprise customers?

**Comment by tasn** [comment, marked as answer]
These are two separate things:
1. The retention period on the Pro plan refers to the max payload retention time. The metadata of the messages remains for longer than that.
2. The 90-day mention in this API is about how much data you're allowed to search in one query. All tiers allow you get data older than 90 days old, you just need to use the before/after parameter. Though as mention in the above point, the payload won't be around for more than the define retention period.

In the Enterprise we support defining custom payload retention. So if you want retention that's longer than 90 days, we can customize your plan to support that.

I hope this helps.

**Reply by dariom** [reply]
Thanks for the explanation.

Following that, two more questions:

1. When retrieving a message that older than 90 days, the payload in the response is `null`?
2. Is it possible to list more than 90 days of data by specifying `before` AND `after` parameters? We don't have 90 days' data yet, so I can't test this!

**Reply by dariom** [reply]
A friendly bump in case my follow-up question was missed. Let me know if I should rather start a separate discussion.

**Reply by tasn** [reply]
1. Yes.
2. Yup! The 90 days limit is a query window, but you can query for 90 days of data from 6 months ago for example.
